### PR TITLE
Fix bug in bond joins

### PR DESCRIPTION
**Pull request by DavidACosgrove**
Not all the bond joins were being smoothed properly. 
#### Reference Issue
No issue.
I spotted it with test23_1.svg from test1.cpp, so that's where I've put the extra tests.  It was a pretty universal problem, though.

#### What does this implement/fix? Explain your changes.
in DrawMol.cpp, singleBondLines_ wasn't being cleared before a re-scale and the leftovers were confusing the code that worked out which bits to put a mitre round.

#### Any other comments?
This bug fix is relevant to the ACS1996 code #5425, not the current drawing code.

### CODE REVIEW

File: Code/GraphMol/MolDraw2D/test1.cpp [hunk 1]
```diff
@@ -4533,6 +4533,12 @@ void test23JSONAtomColourPalette() {
     TEST_ASSERT(text.find("fill:#3366E5") != std::string::npos);
     TEST_ASSERT(text.find("fill:#E59900") != std::string::npos);
 #endif
+    std::regex regex(
+        R"regex(path d='M \d+.\d+,\d+.\d+ L \d+.\d+,\d+.\d+ L \d+.\d+,\d+.\d+.*)regex");
```
Comment (greglandrum) [inline Code/GraphMol/MolDraw2D/test1.cpp:4537]: I don't think you need the `*` at the end of this, but I'm certainly no regexpert

File: Code/GraphMol/MolDraw2D/test1.cpp [hunk 2]
```diff
@@ -4533,6 +4533,12 @@ void test23JSONAtomColourPalette() {
...
+    std::ptrdiff_t const match_count(
+        std::distance(std::sregex_iterator(text.begin(), text.end(), regex),
+                      std::sregex_iterator()));
```
Comment (greglandrum) [inline Code/GraphMol/MolDraw2D/test1.cpp:4540]: might as well hide a little bit of the horribleness:
```suggestion
    auto match_count = 
        std::distance(std::sregex_iterator(text.begin(), text.end(), regex),
                      std::sregex_iterator());
```

**Review by greglandrum — COMMENTED** [review]
Two comments. If you'd rather leave things as they are, I'm happy to merge as-is.

File: Code/GraphMol/MolDraw2D/test1.cpp (hunk 1 above)
Comment (DavidACosgrove) [reply Code/GraphMol/MolDraw2D/test1.cpp:4537]: I guess it might even be unhelpful.

File: Code/GraphMol/MolDraw2D/test1.cpp (hunk 2 above)
Comment (DavidACosgrove) [reply Code/GraphMol/MolDraw2D/test1.cpp:4540]: I have mixed feelings about auto - it's not always clear what exactly you're getting and how to use it subsequently,  particularly in the case of pointers and references where the downstream syntax is different.  However, as you allude to in this case, std:::ptrdiff_t is hardly revealing, either.  CLion translates it as 'long'.

Comment (greglandrum) [reply Code/GraphMol/MolDraw2D/test1.cpp]: Agreed about auto sometimes making things more difficult. I definitely would be a lot less enthusiastic about it if I didn't use an IDE

**Review by greglandrum — APPROVED** [review]
LGTM

**Pull request merged**